internal/report/store.go

```diff
@@ -0,0 +1,121 @@
+package report
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+)
+
+// PolicyReportStore caches the latest version of PolicyReports
+type policyReportStore struct {
+	// store is a map of namespaces and namespaced PolicyReports
+	store map[string]RInterface
+	// clusterPR is the sole ClusterPolicyReport
+	clusterPR ClusterPolicyReport
+
+	rwm *sync.RWMutex
```

Comment:
I would go with something like that:

```golang
namespacePolicyReports: make(map[string]PolicyReport)
clusterPolicyReport ClusterPolicyRepor

namespacePolicyReportsMutex *sync.RWMutex
clusterPolicyReportsMutex *sync.RWMutex
```

Also, I would remove the `RInterface`. I don't see (yet) why we should benefit from that

Reply:
That's better indeed. Done in https://github.com/kubewarden/audit-scanner/pull/40/commits/0329296a7ee8c77e295d16a18d64d32492974eee.